Normalizing net_pkt hexdump

tbursztyka opened the pull request.

Commit:
net/utils: Make net_hexdump_frags being able to print ll reserve

If requested only and with colors with right SYS_LOG Kconfig options.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ieee802154: Use common net_hexdump_frags instead of local function

Now that net_hexdump_frags can print out the link-layer reserve part,
let's use it.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

andrewboie commented on tests/net/ieee802154/fragment/src/main.c:
```diff
@@ -29,7 +29,7 @@
 #include "net_private.h"
 
-#define DEBUG 0
+#define DEBUG 1
```
was this intentional?

tbursztyka replied:
It was for testing the patch, but that should not be here indeed.

andrewboie commented on tests/net/rpl/src/main.c:
```diff
@@ -150,7 +150,7 @@ static int tester_send(struct net_if *iface, struct net_pkt *pkt)
 	DBG("pkt %p to be sent len %lu\n", pkt, net_pkt_get_len(pkt));
 
 #if 0
```
why do we have an #if 0
make it in terms of some macro, or delete the whole thing

tbursztyka replied:
I don't know the background why it is like that, I think the #if can be removed here. Will do that in another patch to make sure net_hexdump_frags() prints out things only when CONFIG_NET_DEBUG_RPL is enabled.

jukkar replied:
Please do not remove the "#if 0" from this rpl test program. These were there on purpose as some of the tests need more TLC but there has not been time to fix/enhance them.

tbursztyka replied:
I am not removing the #if 0 at line 535, there the comment specifically says it should stay and need more work. It's not the same issue.

andrewboie commented on subsys/net/ip/l2/ieee802154/ieee802154.c:
```diff
@@ -32,53 +32,13 @@
 
 #if 0
```
I know it's outside the patch, but having commented-out code like this is not good.
Make this in terms of some macro IMO

tbursztyka replied:
A Kconfig option will do yes.

tyrantlizard77 reviewed (changes requested):
comments inline

jukkar commented on subsys/net/ip/net_private.h:
```diff
@@ -214,14 +223,37 @@ static inline void net_hexdump(const char *str, const u8_t *packet,
+		SYS_LOG_DBG("%s zero-length packet", str);
+		return;
+	}
+
```
Print header (the str variable) here too.

tbursztyka replied:
indeed

Commit:
ieee802154: Make packet display enablement via a Kconfig option

Such option should be use carefully. Printing out in/out packets is
extremely verbose.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar commented:
Ping @andrewboie are you ok with current version?

jukkar commented:
@andrewboie Ping, we need your approval in order to merge this one.
